### PR TITLE
Remove Optional/build and Optional/fold

### PR DESCRIPTION
**Pull request by sjakobi**
This implements phase 3 of the deprecation plan at

https://docs.dhall-lang.org/howtos/migrations/Deprecation-of-Optional-fold-and-Optional-build.html

Corresponding change to the Haskell implementation: https://github.com/dhall-lang/dhall-haskell/pull/1802

### CODE REVIEW

File: docs/tutorials/string-matrix.md [hunk 1]
```diff
@@ -335,4 +335,4 @@ in  generate-test-matrix
 
 > **Note**: Using the `List/head` function the `generate-test-matrix` could
 > be improved to take a single list of argument.
-> A follow-up tutorial introduces that using `Optional/fold`
+> A follow-up tutorial introduces that using **TODO**.
```
Comment (sjakobi) [inline docs/tutorials/string-matrix.md:338]: @TristanCacqueray I believe you were referring to a yet unwritten tutorial here. Can you make a suggestion how to update this?

File: docs/howtos/migrations/Deprecation-of-Optional-fold-and-Optional-build.md [hunk 2]
```diff
@@ -52,7 +52,7 @@ Users of `Optional/build` should simply use `Some` and `None` directly.
 * Phase 3 - Remove the `Optional/fold` and `Optional/fold` builtins
 
   * Standard version: 17.0.0
-  * Haskell implementation version: To be determined - probably 1.33.0
+  * Haskell implementation version: 1.33.0
   * Ruby implementation version: To be determined
   * Rust implementation version: To be determined
```
Comment (sjakobi) [inline docs/howtos/migrations/Deprecation-of-Optional-fold-and-Optional-build.md:57]: @singpolyma, @Nadrieril If you already know in what version you'll implement this, please let me know. I'd also like to update the version info for Phase 1 above.

I can add other implementations too, if anyone wants theirs included.

File: docs/tutorials/string-matrix.md (hunk 1 above)
Comment (TristanCacqueray) [reply docs/tutorials/string-matrix.md:338]: I guess the new `merge` syntax to consume Optional is so much easier, this no longer deserve it's own tutorial :-) I suggest to remove the last line for now.

File: tests/parser/success/builtinsA.dhall [hunk 3]
```diff
@@ -1,32 +1,35 @@
...
-  , field13 = List/last
-  , field14 = List/indexed
-  , field15 = List/reverse
-  , field16 = Optional/fold
-  , field17 = Optional/build
-  , field18 = True
-  , field19 = False
-  , field20 = None
-  }
+[ Natural/fold
```
Comment (sjakobi) [inline tests/parser/success/builtinsA.dhall:1]: I noticed that this test didn't include some recent additions like `Integer/clamp`, so I regenerated it from the `builtins` rule in `dhall.abnf` and added a reminder there.

File: docs/tutorials/string-matrix.md (hunk 1 above)
Comment (sjakobi) [reply docs/tutorials/string-matrix.md]: Done! Thanks!

File: docs/howtos/migrations/Deprecation-of-Optional-fold-and-Optional-build.md (hunk 2 above)
Comment (Nadrieril) [reply docs/howtos/migrations/Deprecation-of-Optional-fold-and-Optional-build.md:57]: For phase 1 it was 0.3.0. For removal, I'd say 0.6.0

**Pull request merged**